core/src/test/java/com/linecorp/armeria/internal/server/annotation/ResponseConverterFunctionSelectorTest.java

```diff
@@ -0,0 +1,83 @@
+package com.linecorp.armeria.internal.server.annotation;
+
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.annotation.ResponseConverterFunction;
+
+@SuppressWarnings("ConstantConditions")
+class ResponseConverterFunctionSelectorTest {
+
+    private static final ServiceRequestContext ctx = ServiceRequestContext.builder(
+            HttpRequest.of(HttpMethod.GET, "/")).build();
+
+    @Test
+    void prioritisesSpiDelegatingResponseConverterProvider() throws Exception {
+        final ResponseConverterFunction converterFunction =
+                ResponseConverterFunctionSelector.responseConverter(
+                        TestClassWithDelegatingResponseConverterProvider.class,
+                        Collections.singletonList(new MyResponseConverterFunction())
+                );
+
+        final HttpResponse response = converterFunction.convertResponse(ctx, null,
+                                                                        new TestClassWithDelegatingResponseConverterProvider(),
```

Comment:
`TestClassWithDelegatingResponseConverterProvider` doesn't do anything. How about just passing null to reduce the cognitive load of reviewers?
```java
final HttpResponse response = converterFunction.convertResponse(
        ctx, responseHeaders, null, HttpHeaders.of());

```

Reply:
Hey,

`TestClassWithDelegatingResponseConverterProvider` is used in conjunction with `TestDelegatingSpiConverterProvider`, which we've written to produce the response `testDelegatingResponse`

This test ensures that, given the class `TestClassWithDelegatingResponseConverterProvider` is the desired return type, and given that the object to be converted is an instance of `TestClassWithDelegatingResponseConverterProvider`,  the corresponding Spi delegating provider (`TestDelegatingSpiConverterProvider`) is used to convert it, which we assert for using the result `testDelegatingResponse`

If we directly put in null, the test will fail.

Reply:
Please do let me know if I misunderstood your suggestion!

Reply:
Oops, that's correct. I misread the code. 😅